proxy/proxy/client.go

```diff
@@ -342,10 +346,18 @@ func (c *Client) cachedCfg(ctx context.Context, instance string) (string, *tls.C
 		if c.cfgCache == nil {
 			c.cfgCache = make(map[string]cacheEntry)
 		}
+		if c.limiters == nil {
+			c.limiters = make(map[string]*rate.Limiter)
+		}
 		// the state may have changed between critical sections, so double check
 		e = c.cfgCache[instance]
+		limiter := c.limiters[instance]
+		if limiter == nil {
+			limiter = rate.NewLimiter(rate.Every(throttle), 1)
+			c.limiters[instance] = limiter
+		}
 		if needsRefresh(e, refreshCfgBuffer) {
-			if time.Since(e.lastRefreshed) >= throttle {
+			if limiter.Allow() {
 				// start a new refresh and update the cachedEntry to reflect that
 				e.done = c.startRefresh(instance, refreshCfgBuffer)
 				e.lastRefreshed = time.Now()
```

Comment:
We could remove this value as it's only used in the logging message.

Reply:
I'm okay with it being in the logging message.